org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/CleanUpRegistry.java

```diff
@@ -94,10 +101,27 @@ public List<TextEdit> getEditsForAllActiveCleanUps(TextDocumentIdentifier textDo
 
 		// build the context after setting the compiler options so that the built AST has all the required markers
 		CleanUpContextCore context = CleanUpUtils.getCleanUpContext(textDocumentId, opts, monitor);
-
 		List<TextEdit> textEdits = new ArrayList<>();
-		for (ISimpleCleanUp cleanUp : cleanUpsToRun) {
-			textEdits.addAll(CleanUpUtils.getTextEditFromCleanUp(cleanUp, context, monitor));
+		ICompilationUnit cu = context.getCompilationUnit();
+
+		try {
+			for (ISimpleCleanUp cleanUp : cleanUpsToRun) {
+				org.eclipse.text.edits.TextEdit jdtEdit = CleanUpUtils.getTextEditFromCleanUp(cleanUp, context, monitor);
+				if (jdtEdit != null) {
+					cu.applyTextEdit(jdtEdit, monitor);
+					context = CleanUpUtils.getCleanUpContext(cu, opts, monitor);
+				}
+			}
+			// https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textEditArray
+			// Cleanups may have overlapping text edits but LSP does not support this
+			// Generate text edit as the entire document
+			IBuffer cuBuff = cu.getBuffer();
+			String newText = cuBuff.getContents();
+			TextEdit te = new TextEdit(JDTUtils.toRange(cu, 0, cuBuff.getLength()), newText);
```

Comment:
... and this is wrong. I need to use the original compilation unit when crafting the text edit.